weakmap.c

```diff
@@ -418,10 +428,12 @@ wmap_aset_replace(st_data_t *key, st_data_t *val, st_data_t new_key_ptr, int exi
         RUBY_ASSERT(*(VALUE *)*key == new_key);
     }
     else {
-        VALUE *pair = xmalloc(sizeof(VALUE) * 2);
+        struct weakmap_entry *entry = xmalloc(sizeof(struct weakmap_entry));
+
+        RUBY_ASSERT((void *)entry == (void *)&entry->key);
```

Comment:
This expression equals `offsetof(struct weakmap_entry, key) == 0`.
Does this assertion have a meaning?